design/specs/visual-query-builder.md

```diff
@@ -0,0 +1,174 @@
+
+## 2. Creating the Input Table
+
+The input table is created by selecting columns from the base table or creating new ones via formulas.
+
+These columns are added to the result table based on the selection order. Users can rename columns as they wish.
+
+By default, the system will name columns to indicate their source table and column according to the following [naming convention](#column-naming-convention).
+
+Some options might be available for columns that link to multiple records, like filters and aggregations to help users define which data they want to show and how. The system will automatically detect the type of columns and present the available options to users.
+
+### 2.1. Adding Input Columns
+
+To start adding columns, the user clicks on the `Add Column` button located in the `Columns` section on the query configuration panel. This action will display the `Column Selector` component in the inspector panel.
+
+The `Column Selector` will list all available columns, and those in use, when applicable. Additionally there will be a list of all formulas.
+
+The columns in `Column Selector` are listed in a hierarchical structure based on the links that exist between the base table and other tables, in both directions. Icons indicate whether the linked records are single or multiple.
+
+Links are presented as expandable sections containing the columns from the linked tables.
+
+When adding input columns, the user can add them using the list controls or drop the columns directly into the result table.
+
+---
+Wireframes
+
+[Adding Columns](https://share.balsamiq.com/c/oyxTXxqSh8rLqU3JY71DWY.png)
+[Added Column](https://share.balsamiq.com/c/7U9yahZtYWX2G5obkyivoz.png)
+
+### 2.2 Adding Formulas
+
+While adding columns, the user can choose to create a new column by using formulas. Available formulas will be included as part of the column selector, and dropping them into the result table will prompt users to set the formula settings for the selected formula.
+
+Depending on the selected formula, different settings will be available. Formulas will use the column selector component for settings that require the selection of columns. The list of columns will be populated based on the selected formulas.
+
+However, the system only lists those with data types accepted by the formula.
+
+---
+Wireframes
+
+[Added Formula](https://share.balsamiq.com/c/vWfJ9sUYWJxed5Zo5WtPGi.png)
+
+### 2.3 Filtering Input Column Values
+
+Columns that link to multiple records can have filters applied to them in order to retrieve only values that match user-specified criteria. Multiple filters are allowed for each input column.
+
+Filter options will be determined by the data type of the input column.
+
+Filters can be added by clicking on the `Add Filter` option from the query configuration panel or directly from the results table column header menu.
+
+---
+Wireframes
+
+[Added Input Filter](https://share.balsamiq.com/c/tK2hZy6FB4zVYpN56ejpBk.png)
```

Comment:
Looks good. Nitpick: The column name in the table and inspector are different (Total Actors vs. Actors)